### PR TITLE
Revise RawResource sql constraint

### PR DESCRIPTION
**Pull request by v-mserdar**
## Description
Addresses the following sql error.
--Msg 8604, Level 16, State 1, Line 35

--ALTER TABLE SWITCH statement failed. Table 'My_Resource_2' has a column level check constraint 'CK__My_Resour__RawRe__278EDA44' on column 'RawResource' that is not loadable for semantic validation.

## Related issues
Addresses [94498](https://microsofthealth.visualstudio.com/Health/_workitems/edit/94498).

## Testing
Ran this locally by starting up SqlServer R4 in debug mode. My initial SchemaVersion was 37. After getting this running my version went to 38 then ended at 39 with the correct constraint applied to the Resource table.

## FHIR Team Checklist
- [X] **Update the title** of the PR to be succinct and less than 50 characters
- [X] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [X] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [ ] Tag the PR with Azure API for FHIR if this will release to the Azure API for FHIR managed service (CosmosDB or common code related to service)
- [X] Tag the PR with Azure Healthcare APIs if this will release to the Azure Healthcare APIs managed service (Sql server or common code related to service)
- [ ] CI is green before merge
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch (reason)

### CODE REVIEW

File: src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/39.diff.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿/*************************************************************
+    This migration revises an existing constraint for the data length of the rawresource column
+**************************************************************/
+
+IF  EXISTS (SELECT * FROM sys.objects WHERE name = N'CH_Resource_RawResource_Length' AND type = N'C')
+    ALTER TABLE dbo.Resource DROP CONSTRAINT CH_Resource_RawResource_Length
+GO
+
+ALTER TABLE dbo.Resource WITH NOCHECK
+    ADD CONSTRAINT CH_Resource_RawResource_Length CHECK (DATALENGTH(RawResource) > 0x0)
```
Comment (SergeyGaluzo) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/39.diff.sql:10]: > CHECK (DATALENGTH(RawResource) > 0x0) [](http://example.com/codeflow?start=49&length=38)

It is incorrect. It should be (RawResource > 0x0)

**Review by SergeyGaluzo — CHANGES_REQUESTED** [review]
:clock1:

File: src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Tables/Resource.sql [hunk 2]
```diff
@@ -12,7 +12,7 @@
     SearchParamHash             varchar(64)             NULL,
 
     CONSTRAINT PKC_Resource PRIMARY KEY CLUSTERED (ResourceTypeId, ResourceSurrogateId) WITH (DATA_COMPRESSION = PAGE) ON PartitionScheme_ResourceTypeId(ResourceTypeId),
-    CONSTRAINT CH_Resource_RawResource_Length CHECK (DATALENGTH(RawResource) > 0)
+    CONSTRAINT CH_Resource_RawResource_Length CHECK (DATALENGTH(RawResource) > 0x0)
```
Comment (SergeyGaluzo) [inline src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Tables/Resource.sql:15]: >DATALENGTH(RawResource) > 0x0) [](http://example.com/codeflow?start=53&length=30)

Incorrect.

Comment (v-mserdar) [reply src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Tables/Resource.sql]: Also fixed in updated commit.

**Review by SergeyGaluzo — APPROVED** [review]
:shipit:

**Review by feordin — APPROVED** [review]

**Pull request merged**